commit 846cdd36c94702999b907d8a33b6e8bc4c37b5a0
Author: Nicolas CHARLES <nicolas.charles@normation.com>
Date:   Tue Sep 16 12:00:08 2014 +0200

    Fixes #2718: correct condition for disk storage computation

diff --git a/lib/FusionInventory/Agent/Task/Inventory/Linux/Storages.pm b/lib/FusionInventory/Agent/Task/Inventory/Linux/Storages.pm
index 618749f..6341f00 100644
--- a/lib/FusionInventory/Agent/Task/Inventory/Linux/Storages.pm
+++ b/lib/FusionInventory/Agent/Task/Inventory/Linux/Storages.pm
@@ -7,6 +7,7 @@ use English qw(-no_match_vars);
 
 use FusionInventory::Agent::Tools;
 use FusionInventory::Agent::Tools::Linux;
+use FusionInventory::Agent::Tools::Unix;
 
 sub isEnabled {
     return 1;
@@ -93,7 +94,7 @@ sub doInventory {
             );
         }
 
-        if ($device->{DISKSIZE} && $device->{TYPE} =~ /^cd/) {
+        if (!$device->{DISKSIZE} && $device->{TYPE} !~ /^cd/) {
             $device->{DISKSIZE} = getDeviceCapacity(device => '/dev/' . $device->{NAME});
         }
 
