--- jetty7/bin/jetty.sh	2011-09-12 11:07:16.913497607 +0200
+++ jetty7/bin/jetty.sh	2011-09-12 12:15:32.633499423 +0200
@@ -408,6 +408,29 @@
   echo "RUN_CMD        =  ${RUN_CMD}"
 fi
 
+#####################################################
+# Check rudder plugins before action
+#####################################################
+PLUGINSDIR='/opt/rudder/jetty7/rudder-plugins/'
+CHKPLUGIN=`ls $PLUGINSDIR | wc -l`
+PLUGINSPATH=""
+
+CPT=0
+if [ $CHKPLUGIN -gt 0 ]
+then
+    for fic in $PLUGINSDIR*
+    do
+        if [ $CPT -eq 0 ]
+        then
+            PLUGINSPATH=$fic
+        else
+            PLUGINSPATH="$PLUGINSPATH:$fic"
+        fi
+        CPT=$(($CPT+1))
+    done
+sed -i "s%extraClasspath\">.*</S%extraClasspath\">$PLUGINSPATH</S%" /opt/rudder/jetty7/contexts/rudder.xml
+fi
+
 ##################################################
 # Do the action
 ##################################################
