--- jetty7/bin/jetty.sh.orig	2014-03-13 01:14:20.518910880 +0100
+++ jetty7/bin/jetty.sh	2014-03-13 01:15:31.630682858 +0100
@@ -173,11 +173,10 @@
   ETC=$HOME/etc
 fi
 
-for CONFIG in $ETC/default/jetty{,7} $HOME/.jettyrc; do
-  if [ -f "$CONFIG" ] ; then 
-    readConfig "$CONFIG"
-  fi
-done
+CONFIG=/etc/default/rudder-jetty
+if [ -f "$CONFIG" ] ; then 
+  readConfig "$CONFIG"
+fi
 
 
 ##################################################
