--- jetty7/bin/jetty-sles.sh 2010-12-06 19:26:42.000000000 +0100
+++ jetty7/bin/jetty-sles.sh 2011-07-27 17:39:27.866582165 +0200
@@ -443,7 +443,8 @@
       exit
     fi
 
-    if type start-stop-daemon > /dev/null 2>&1 
+    #if type start-stop-daemon > /dev/null 2>&1 
+	if [ "$START_STOP_DAEMON" = "1" ] && type start-stop-daemon > /dev/null 2>&1
     then
       unset CH_USER
       if [ -n "$JETTY_USER" ]
@@ -497,7 +498,8 @@
 
   stop)
     echo -n "Stopping Jetty: "
-    if type start-stop-daemon > /dev/null 2>&1; then
+    #if type start-stop-daemon > /dev/null 2>&1; then
+	if [ "$START_STOP_DAEMON" = "1" ] && type start-stop-daemon > /dev/null 2>&1; then
       start-stop-daemon -K -p"$JETTY_PID" -d"$JETTY_HOME" -a "$JAVA" -s HUP
       
       TIMEOUT=30
