--- jetty7/bin/jetty.sh.orig	2014-03-13 00:55:51.261843139 +0100
+++ jetty7/bin/jetty.sh	2014-03-13 00:56:57.505715067 +0100
@@ -431,6 +431,8 @@
 ##################################################
 case "$ACTION" in
   start)
+    UMASK="0007"
+    echo "Setting umask to ${UMASK}"
     echo -n "Starting Jetty: "
 
     if (( NO_START )); then 
@@ -446,10 +448,11 @@
         CH_USER="-c$JETTY_USER"
       fi
 
-      start-stop-daemon -S -p"$JETTY_PID" $CH_USER -d"$JETTY_HOME" -b -m -a "$JAVA" -- "${RUN_ARGS[@]}" --daemon
+      start-stop-daemon -k ${UMASK} -S -p"$JETTY_PID" $CH_USER -d"$JETTY_HOME" -b -m -a "$JAVA" -- "${RUN_ARGS[@]}" --daemon
 
     else
 
+      umask ${UMASK}
       if [ -f "$JETTY_PID" ]
       then
         if running $JETTY_PID
