--- jetty7/bin/jetty.sh	2014-02-25 14:30:19.343794595 +0100
+++ jetty7/bin/jetty.sh	2014-02-25 14:50:45.732183652 +0100
@@ -11,7 +11,7 @@
 
 # Configuration files
 #
-# /etc/default/jetty
+# /etc/default/rudder-jetty
 #   If it exists, this is read at the start of script. It may perform any 
 #   sequence of shell commands, like setting relevant environment variables.
 #
@@ -149,12 +149,9 @@
 ##################################################
 # Read any configuration files
 ##################################################
-for CONFIG in /etc/default/jetty{,7} $HOME/.jettyrc; do
-  if [ -f "$CONFIG" ] ; then 
-    readConfig "$CONFIG"
-  fi
-done
-
+if [ -f "/etc/default/rudder-jetty" ] ; then 
+  readConfig "/etc/default/rudder-jetty"
+fi
 
 ##################################################
 # Set tmp if not already set.
