diff -upr ./openldap-source/servers/slapd/back-mdb/filterindex.c openldap-source-2/servers/slapd/back-mdb/filterindex.c
--- ./openldap-source/servers/slapd/back-mdb/filterindex.c	2016-02-06 00:57:45.000000000 +0100
+++ openldap-source-2/servers/slapd/back-mdb/filterindex.c	2017-03-16 14:28:29.411279600 +0100
@@ -734,7 +734,7 @@ equality_candidates(
 		&dbi, &mask, &prefix );
 
 	if ( rc == LDAP_INAPPROPRIATE_MATCHING ) {
-		Debug( LDAP_DEBUG_ANY,
+		Debug( LDAP_DEBUG_TRACE,
 			"<= mdb_equality_candidates: (%s) not indexed\n", 
 			ava->aa_desc->ad_cname.bv_val, 0, 0 );
 		return 0;
@@ -850,7 +850,7 @@ approx_candidates(
 		&dbi, &mask, &prefix );
 
 	if ( rc == LDAP_INAPPROPRIATE_MATCHING ) {
-		Debug( LDAP_DEBUG_ANY,
+		Debug( LDAP_DEBUG_TRACE,
 			"<= mdb_approx_candidates: (%s) not indexed\n",
 			ava->aa_desc->ad_cname.bv_val, 0, 0 );
 		return 0;
@@ -969,7 +969,7 @@ substring_candidates(
 		&dbi, &mask, &prefix );
 
 	if ( rc == LDAP_INAPPROPRIATE_MATCHING ) {
-		Debug( LDAP_DEBUG_ANY,
+		Debug( LDAP_DEBUG_TRACE,
 			"<= mdb_substring_candidates: (%s) not indexed\n",
 			sub->sa_desc->ad_cname.bv_val, 0, 0 );
 		return 0;
@@ -1085,7 +1085,7 @@ inequality_candidates(
 		&dbi, &mask, &prefix );
 
 	if ( rc == LDAP_INAPPROPRIATE_MATCHING ) {
-		Debug( LDAP_DEBUG_ANY,
+		Debug( LDAP_DEBUG_TRACE,
 			"<= mdb_inequality_candidates: (%s) not indexed\n", 
 			ava->aa_desc->ad_cname.bv_val, 0, 0 );
 		return 0;
