From ed90c7614f9efeef592c6d8f76018669774f7f4a Mon Sep 17 00:00:00 2001
From: Matthew Cattell <matthew.cattell@cfengine.com>
Date: Thu, 16 Jan 2014 15:43:26 +0100
Subject: [PATCH] partially revert a change intended to alter the log message -
 fixes 4038

---
 cf-agent/cf-agent.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/cf-agent/cf-agent.c b/cf-agent/cf-agent.c
index 61ca750..18a76da 100644
--- a/cf-agent/cf-agent.c
+++ b/cf-agent/cf-agent.c
@@ -1391,13 +1391,12 @@ static void KeepAgentPromise(EvalContext *ctx, Promise *pp, ARG_UNUSED void *par
         if (LEGACY_OUTPUT)
         {
             Log(LOG_LEVEL_VERBOSE, ". . . . . . . . . . . . . . . . . . . . . . . . . . . . ");
-            Log(LOG_LEVEL_VERBOSE, "Skipping whole next promise (%s), as var-context %s is not relevant", pp->promiser,
-                  sp);
+            Log(LOG_LEVEL_VERBOSE, "Skipping whole next promise (%s), as var-context %s is not relevant", pp->promiser, sp ? sp : pp->classes);
             Log(LOG_LEVEL_VERBOSE, ". . . . . . . . . . . . . . . . . . . . . . . . . . . . ");
         }
         else
         {
-            Log(LOG_LEVEL_VERBOSE, "Skipping next promise '%s', as var-context '%s' is not relevant", pp->promiser, sp);
+            Log(LOG_LEVEL_VERBOSE, "Skipping next promise '%s', as var-context '%s' is not relevant", pp->promiser, sp ? sp : pp->classes);    
         }
         return;
     }
-- 
1.8.1.5

