diff -r -auN cfengine-3.6.0/cf-monitord/mon_processes.c cfengine-3.6.0-patch1/cf-monitord/mon_processes.c
--- cfengine-3.6.0/cf-monitord/mon_processes.c	2014-04-01 16:41:01.000000000 +0200
+++ cfengine-3.6.0-patch1/cf-monitord/mon_processes.c	2014-09-03 14:49:50.964464893 +0200
@@ -74,7 +74,7 @@
     char pscomm[CF_BUFSIZE];
     char user[CF_MAXVARSIZE];
 
-    snprintf(pscomm, CF_BUFSIZE, "%s %s", VPSCOMM[VSYSTEMHARDCLASS], VPSOPTS[VSYSTEMHARDCLASS]);
+    snprintf(pscomm, CF_BUFSIZE, "%s %s", VPSCOMM[VPSHARDCLASS], VPSOPTS[VPSHARDCLASS]);
 
     if ((pp = cf_popen(pscomm, "r", true)) == NULL)
     {
diff -r -auN cfengine-3.6.0/libenv/sysinfo.c cfengine-3.6.0-patch1/libenv/sysinfo.c
--- cfengine-3.6.0/libenv/sysinfo.c	2014-06-11 16:15:15.000000000 +0200
+++ cfengine-3.6.0-patch1/libenv/sysinfo.c	2014-09-03 14:53:07.179129593 +0200
@@ -470,6 +470,7 @@
                     found = true;
 
                     VSYSTEMHARDCLASS = (PlatformContext) i;
+                    VPSHARDCLASS = (PlatformContext) i; /* this one can be overriden at vz detection */
                     EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "class", CLASSTEXT[i], CF_DATA_TYPE_STRING, "inventory,source=agent,attribute_name=OS type");
                     break;
                 }
@@ -2310,22 +2311,32 @@
 #define OPENVZ_HOST_FILENAME "/proc/bc/0"
 #define OPENVZ_GUEST_FILENAME "/proc/vz"
 /* path to the vzps binary */
-#define OPENVZ_VZPS_FILE "/bin/vzps"
+#define OPENVZ_VZPS_FILE "/opt/rudder/bin/vzps.py"
     struct stat statbuf;
+    int i;
 
     /* The file /proc/bc/0 is present on host
        The file /proc/vz is present on guest
-       If the host has /bin/vzps, we should use it for checking processes
+       If the host has /opt/rudder/bin/vzps.py, we should use it for checking processes
     */
 
     if (stat(OPENVZ_HOST_FILENAME, &statbuf) != -1)
     {
         Log(LOG_LEVEL_VERBOSE, "This appears to be an OpenVZ/Virtuozzo/Parallels Cloud Server host system.\n");
         EvalContextClassPutHard(ctx, "virt_host_vz", "inventory,attribute_name=Virtual host,source=agent");
-        /* if the file /bin/vzps is there, it is safe to use the processes promise type */ 
+        /* if the file /opt/rudder/bin/vzps.py is there, it is safe to use the processes promise type */ 
         if (stat(OPENVZ_VZPS_FILE, &statbuf) != -1)
         {
             EvalContextClassPutHard(ctx, "virt_host_vz_vzps", "inventory,attribute_name=Virtual host,source=agent");
+            /* here we must redefine the value of VPSHARDCLASS */
+            for (i = 0; i < PLATFORM_CONTEXT_MAX; i++)
+            {
+                if (!strcmp(CLASSATTRIBUTES[i][0], "virt_host_vz_vzps"))
+                {
+                   VPSHARDCLASS = (PlatformContext) i;
+                   break;
+                }
+            }
         }
         else
         {
diff -r -auN cfengine-3.6.0/libpromises/cf3.extern.h cfengine-3.6.0-patch1/libpromises/cf3.extern.h
--- cfengine-3.6.0/libpromises/cf3.extern.h	2014-04-01 16:41:01.000000000 +0200
+++ cfengine-3.6.0-patch1/libpromises/cf3.extern.h	2014-09-03 14:49:50.964464893 +0200
@@ -47,6 +47,7 @@
 
 extern char VDOMAIN[CF_MAXVARSIZE];
 extern PlatformContext VSYSTEMHARDCLASS;
+extern PlatformContext VPSHARDCLASS; /* used to define which ps command to use*/
 extern char VFQNAME[];
 extern char VUQNAME[];
 
diff -r -auN cfengine-3.6.0/libpromises/cf3globals.c cfengine-3.6.0-patch1/libpromises/cf3globals.c
--- cfengine-3.6.0/libpromises/cf3globals.c	2014-04-01 16:41:01.000000000 +0200
+++ cfengine-3.6.0-patch1/libpromises/cf3globals.c	2014-09-03 14:49:50.964464893 +0200
@@ -161,3 +161,4 @@
    call external utility
 */
 PlatformContext VSYSTEMHARDCLASS; /* GLOBAL_E?, initialized_later */
+PlatformContext VPSHARDCLASS; /* used to define which ps command to use*/
diff -r -auN cfengine-3.6.0/libpromises/classes.c cfengine-3.6.0-patch1/libpromises/classes.c
--- cfengine-3.6.0/libpromises/classes.c	2014-06-11 16:13:56.000000000 +0200
+++ cfengine-3.6.0-patch1/libpromises/classes.c	2014-09-03 14:54:10.346708558 +0200
@@ -50,7 +50,7 @@
 const char *const VPSCOMM[] =
 {
     [PLATFORM_CONTEXT_UNKNOWN] = "",
-    [PLATFORM_CONTEXT_OPENVZ] = "/bin/vzps",                /* virt_host_vz_vzps */
+    [PLATFORM_CONTEXT_OPENVZ] = "/opt/rudder/bin/vzps.py",                /* virt_host_vz_vzps */
     [PLATFORM_CONTEXT_HP] = "/bin/ps",                  /* hpux */
     [PLATFORM_CONTEXT_AIX] = "/bin/ps",                  /* aix */
     [PLATFORM_CONTEXT_LINUX] = "/bin/ps",                  /* linux */
@@ -76,7 +76,7 @@
 const char *const VPSOPTS[] =
 {
     [PLATFORM_CONTEXT_UNKNOWN] = "",
-    [PLATFORM_CONTEXT_OPENVZ] = "-E 0 -o user,pid,ppid,pgid,pcpu,pmem,vsz,ni,rss,nlwp,stime,time,args",   /* virt_host_vz_vzps (with vzps, the -E 0 replace the -e) */
+    [PLATFORM_CONTEXT_OPENVZ] = "-E 0 -eo user,pid,ppid,pgid,pcpu,pmem,vsz,ni,rss,nlwp,stime,time,args",   /* virt_host_vz_vzps */
     [PLATFORM_CONTEXT_HP] = "-ef",                      /* hpux */
     [PLATFORM_CONTEXT_AIX] =  "-N -eo user,pid,ppid,pgid,pcpu,pmem,vsz,ni,stat,st=STIME,time,args", /* aix */
     [PLATFORM_CONTEXT_LINUX] = "-eo user,pid,ppid,pgid,pcpu,pmem,vsz,ni,rss,nlwp,stime,time,args",   /* linux */
diff -r -auN cfengine-3.6.0/libpromises/processes_select.c cfengine-3.6.0-patch1/libpromises/processes_select.c
--- cfengine-3.6.0/libpromises/processes_select.c	2014-06-03 14:39:00.000000000 +0200
+++ cfengine-3.6.0-patch1/libpromises/processes_select.c	2014-09-03 14:49:50.964464893 +0200
@@ -690,7 +690,7 @@
 
     if (IsGlobalZone())
     {
-        snprintf(psopts, CF_BUFSIZE, "%s,zone", VPSOPTS[VSYSTEMHARDCLASS]);
+        snprintf(psopts, CF_BUFSIZE, "%s,zone", VPSOPTS[VPSHARDCLASS]);
         return psopts;
     }
 
@@ -703,7 +703,7 @@
 
 # endif
 
-    return VPSOPTS[VSYSTEMHARDCLASS];
+    return VPSOPTS[VPSHARDCLASS];
 }
 #endif
 
@@ -768,7 +768,7 @@
 
     const char *psopts = GetProcessOptions();
 
-    snprintf(pscomm, CF_MAXLINKSIZE, "%s %s", VPSCOMM[VSYSTEMHARDCLASS], psopts);
+    snprintf(pscomm, CF_MAXLINKSIZE, "%s %s", VPSCOMM[VPSHARDCLASS], psopts);
 
     Log(LOG_LEVEL_VERBOSE, "Observe process table with %s", pscomm);
 
diff -r -auN cfengine-3.6.0/tests/unit/mon_processes_test.c cfengine-3.6.0-patch1/tests/unit/mon_processes_test.c
--- cfengine-3.6.0/tests/unit/mon_processes_test.c	2014-04-01 16:41:01.000000000 +0200
+++ cfengine-3.6.0-patch1/tests/unit/mon_processes_test.c	2014-09-03 14:49:50.964464893 +0200
@@ -104,10 +104,13 @@
 
 #if defined(__sun)
     VSYSTEMHARDCLASS = PLATFORM_CONTEXT_SOLARIS;
+    VPSHARDCLASS = PLATFORM_CONTEXT_SOLARIS;
 #elif defined(_AIX)
     VSYSTEMHARDCLASS = PLATFORM_CONTEXT_AIX;
+    VPSHARDCLASS = PLATFORM_CONTEXT_AIX;
 #elif defined(__linux__)
     VSYSTEMHARDCLASS = PLATFORM_CONTEXT_LINUX;
+    VPSHARDCLASS = PLATFORM_CONTEXT_LINUX;
 #endif
 
     PRINT_TEST_BANNER();
