--- a/cf-agent/verify_packages.c	2015-05-15 11:05:55.538932003 +0200
+++ b/cf-agent/verify_packages.c	2015-05-15 11:06:57.248786576 +0200
@@ -2589,7 +2589,7 @@
                     }
 
                     PromiseResult result = PROMISE_RESULT_NOOP;
-                    EvalContextStackPushPromiseFrame(ctx, pp, false);
+                    EvalContextStackPushPromiseFrame(ctx, ppi, false);
                     if (EvalContextStackPushPromiseIterationFrame(ctx, 0, NULL))
                     {
                         if (ExecPackageCommand(ctx, command_string, verify, true, a, ppi, &result))
