--- a/cf-agent/verify_packages.c	2015-05-15 11:09:22.603799486 +0200
+++ b/cf-agent/verify_packages.c	2015-05-15 11:09:46.206126337 +0200
@@ -2684,7 +2684,7 @@
                     }
 
                     PromiseResult result = PROMISE_RESULT_NOOP;
-                    EvalContextStackPushPromiseFrame(ctx, pp, false);
+                    EvalContextStackPushPromiseFrame(ctx, ppi, false);
                     if (EvalContextStackPushPromiseIterationFrame(ctx, 0, NULL))
                     {
                         if (ExecPackageCommand(ctx, command_string, verify, true, a, ppi, &result))
