diff -ur a/libcfnet/classic.c b/libcfnet/classic.c
--- a/libcfnet/classic.c	2014-06-03 14:39:00.000000000 +0200
+++ b/libcfnet/classic.c	2015-07-07 09:41:57.058600817 +0200
@@ -79,6 +79,7 @@
             {
                 if (LastRecvTimedOut())
                 {
+                    close(sd);
                     Log(LOG_LEVEL_ERR,
                         "Timeout - remote end did not respond with the expected amount of data "
                         "(received=%d, expecting=%d). (recv: %s)",

