diff -ur a/cf-agent/verify_packages.c b/cf-agent/verify_packages.c
--- a/cf-agent/verify_packages.c	2014-12-05 22:58:49.000000000 +0100
+++ b/cf-agent/verify_packages.c	2015-07-16 14:53:01.704005993 +0200
@@ -1780,7 +1780,8 @@
             }
             else
             {
-                Log(LOG_LEVEL_VERBOSE, "Installed package is up to date, not updating");
+                cfPS_HELPER_1ARG(ctx, LOG_LEVEL_VERBOSE, PROMISE_RESULT_NOOP, pp, a,
+                    "Installed packaged '%s' is up to date, not updating", pp->promiser);
                 break;
             }
         }
