diff -upr cfengine-source/configure cfengine-source-b/configure
--- cfengine-source/configure	2018-02-02 15:56:07.000000000 +0100
+++ cfengine-source-b/configure	2018-04-10 15:35:53.235335127 +0200
@@ -14608,12 +14608,12 @@ if eval test \"x\$"$as_ac_Header"\" = x"
 #define `$as_echo "HAVE_$ac_header" | $as_tr_cpp` 1
 _ACEOF
  libacl_header_found=yes
-else
-  if test "x$with_libacl" != xcheck; then as_fn_error $? "Cannot find libacl header files" "$LINENO" 5; fi
 fi
 
 done
 
+    if test "x$libacl_header_found" != "xyes" && test "x$with_libacl" != xcheck; then as_fn_error $? "Cannot find libacl library headers" "$LINENO" 5; fi
+
 
 
   #
