diff --git a/libntech/libutils/libcrypto-compat.c b/libntech/libutils/libcrypto-compat.c
index bf39f57..5d29b7c 100644
--- a/libntech/libutils/libcrypto-compat.c
+++ b/libntech/libutils/libcrypto-compat.c
@@ -14,7 +14,7 @@
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
 
 #include <string.h>
-#include <openssl/engine.h>
+//#include <openssl/engine.h>
 #include <openssl/bn.h>                                         /* BN_* */
 
 
