diff --git a/cf-agent/verify_files_utils.c b/cf-agent/verify_files_utils.c
index 4ed5284..bb3630c 100644
--- a/cf-agent/verify_files_utils.c
+++ b/cf-agent/verify_files_utils.c
@@ -1375,7 +1375,9 @@ bool CopyRegularFile(EvalContext *ctx, const char *source, const char *dest, con
     else
     {
         // If preserve is true, retain permissions of source file
-        if (attr->copy.preserve)
+	// keep previous behavior
+	if (true)
+        //if (attr->copy.preserve)
         {
             if (!CopyRegularFileDisk(source, new))
             {
