diff --git a/cf-agent/verify_files_utils.c b/cf-agent/verify_files_utils.c
index 7046026..383c69b 100644
--- a/cf-agent/verify_files_utils.c
+++ b/cf-agent/verify_files_utils.c
@@ -1366,7 +1366,9 @@ bool CopyRegularFile(EvalContext *ctx, const char *source, const char *dest, str
     else
     {
         // If preserve is true, retain permissions of source file
-        if (attr.copy.preserve)
+        //if (attr.copy.preserve)
+	// Keep previous behavior to stay compatible
+	if (true)
         {
             if (!CopyRegularFileDisk(source, new))
             {
