diff --color -ruN cfengine-core.old/libntech/libutils/logging.c cfengine-core/libntech/libutils/logging.c
--- cfengine-core.old/libntech/libutils/logging.c	2022-08-10 09:24:55.219272530 +0200
+++ cfengine-core/libntech/libutils/logging.c	2022-08-10 09:25:10.535277405 +0200
@@ -32,6 +32,7 @@
 #include <definitions.h>        /* CF_BUFSIZE */
 
 char VPREFIX[1024] = ""; /* GLOBAL_C */
+bool USE_SYSLOG = true;
 
 static char AgentType[80] = "generic";
 static bool TIMESTAMPS = false;
@@ -276,8 +277,10 @@
 void LogToSystemLog(const char *msg, LogLevel level)
 {
     char logmsg[4096];
-    snprintf(logmsg, sizeof(logmsg), "CFEngine(%s) %s %s\n", AgentType, VPREFIX, msg);
-    syslog(LogLevelToSyslogPriority(level), "%s", logmsg);
+    if(USE_SYSLOG) {
+      snprintf(logmsg, sizeof(logmsg), "CFEngine(%s) %s %s\n", AgentType, VPREFIX, msg);
+      syslog(LogLevelToSyslogPriority(level), "%s", logmsg);
+    }
 }
 #endif  /* !__MINGW32__ */
 
diff --color -ruN cfengine-core.old/libpromises/generic_agent.c cfengine-core/libpromises/generic_agent.c
--- cfengine-core.old/libpromises/generic_agent.c	2022-08-10 09:24:45.743269573 +0200
+++ cfengine-core/libpromises/generic_agent.c	2022-08-10 09:25:31.711284340 +0200
@@ -81,6 +81,8 @@
 #define AUGMENTS_CLASSES_REGULAR_EXPRESSIONS "regular_expressions"
 #define AUGMENTS_COMMENT_KEY "comment"
 
+extern bool USE_SYSLOG;
+
 static pthread_once_t pid_cleanup_once = PTHREAD_ONCE_INIT; /* GLOBAL_T */
 
 static char PIDFILE[CF_BUFSIZE] = ""; /* GLOBAL_C */
@@ -2132,11 +2134,14 @@
     Log(LOG_LEVEL_VERBOSE, "SET Syslog FACILITY = %s", retval);
 
     CloseLog();
-    OpenLog(ParseFacility(retval));
-    SetSyslogFacility(ParseFacility(retval));
-    if (!StoreDefaultLogFacility())
-    {
+    if(strcmp(retval, "NONE")==0) {
+      USE_SYSLOG = false;
+    } else {
+      OpenLog(ParseFacility(retval));
+      SetSyslogFacility(ParseFacility(retval));
+      if (!StoreDefaultLogFacility()) {
         Log(LOG_LEVEL_ERR, "Failed to store default log facility");
+      }
     }
 }
 
diff --color -ruN cfengine-core.old/libpromises/mod_common.c cfengine-core/libpromises/mod_common.c
--- cfengine-core.old/libpromises/mod_common.c	2022-08-10 09:24:45.743269573 +0200
+++ cfengine-core/libpromises/mod_common.c	2022-08-10 09:25:10.535277405 +0200
@@ -48,7 +48,7 @@
 #include <syntax.h>
 
 #define CF_LOGRANGE    "stdout|udp_syslog|(\042?[a-zA-Z]:\\\\.*)|(/.*)"
-#define CF_FACILITY "LOG_USER,LOG_DAEMON,LOG_LOCAL0,LOG_LOCAL1,LOG_LOCAL2,LOG_LOCAL3,LOG_LOCAL4,LOG_LOCAL5,LOG_LOCAL6,LOG_LOCAL7"
+#define CF_FACILITY "LOG_USER,LOG_DAEMON,LOG_LOCAL0,LOG_LOCAL1,LOG_LOCAL2,LOG_LOCAL3,LOG_LOCAL4,LOG_LOCAL5,LOG_LOCAL6,LOG_LOCAL7,NONE"
 
 static const char *const POLICY_ERROR_VARS_CONSTRAINT_DUPLICATE_TYPE =
     "Variable contains existing data type contstraint %s, tried to "
