diff --git a/cf-agent/verify_files_utils.c b/cf-agent/verify_files_utils.c
index 4d1433e..9a308d9 100644
--- a/cf-agent/verify_files_utils.c
+++ b/cf-agent/verify_files_utils.c
@@ -2697,7 +2697,7 @@ static PromiseResult CopyFileSources(EvalContext *ctx, char *destination, const
         else
         {
             cfPS(ctx, LOG_LEVEL_INFO, PROMISE_RESULT_FAIL, pp, attr,
-                 "Can't stat file '%s' on '%s' in files.copy_from promise",
+                 "Can't stat file '%s' on '%s' in files.copy_from promise, it may be missing or access may not be authorized",
                  BufferData(source), conn ? conn->remoteip : "localhost");
             BufferDestroy(source);
             return PROMISE_RESULT_FAIL;
