diff --color -ruN cfengine-source.orig/libpromises/feature.c cfengine-source/libpromises/feature.c
--- cfengine-source.orig/libpromises/feature.c	2023-05-10 12:38:35.192123758 +0200
+++ cfengine-source/libpromises/feature.c	2023-05-10 12:41:33.602000922 +0200
@@ -28,6 +28,7 @@
     "def_json_preparse",
     "host_specific_data_load",
     "copyfrom_restrict_keys",
+    "syslog_disableable",
     NULL
 };
 
