From 35f0e5c207fcbd8b2f636cf612603f03e18e90b5 Mon Sep 17 00:00:00 2001
From: Alexis Mousset <alexis.mousset@rudder.io>
Date: Mon, 18 Sep 2023 21:41:28 +0200
Subject: [PATCH] CFE-4256: Fallback on /usr/bin/getent when /bin/getent does
 not exist.

---
 libpromises/unix.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/libpromises/unix.c b/libpromises/unix.c
index 613146934a..5982d8e396 100644
--- a/libpromises/unix.c
+++ b/libpromises/unix.c
@@ -247,7 +247,16 @@ static bool GetUserGroupInfoFromGetent(const char *type, const char *query,
                                        LogLevel error_log_level)
 {
     char buf[CF_BUFSIZE];
-    NDEBUG_UNUSED int print_ret = snprintf(buf, sizeof(buf), "/bin/getent %s %s", type, query);
+    struct stat sb;
+    char* getent_bin;
+
+    if (stat("/bin/getent", &sb) == 0)
+    {
+        getent_bin = "/bin/getent";
+    } else {
+        getent_bin = "/usr/bin/getent";
+    }
+    NDEBUG_UNUSED int print_ret = snprintf(buf, sizeof(buf), "%s %s %s", getent_bin, type, query);
     assert(print_ret < sizeof(buf));
 
     FILE *out = cf_popen(buf, "r", OUTPUT_SELECT_STDOUT);
