diff --git a/libenv/sysinfo.c b/libenv/sysinfo.c
index 6beb59b..8c80edd 100644
--- a/libenv/sysinfo.c
+++ b/libenv/sysinfo.c
@@ -1181,6 +1181,7 @@ static void OSReleaseParse(EvalContext *ctx, const char *file_path)
                 alias = "redhat";
             }
             else if (StringEqual(os_release_id, "opensuse") || 
+                     StringEqual(os_release_id, "sled") ||
                      StringEqual(os_release_id, "sles"))
             {
                 alias = "suse";
