commit fb9c0d01488750b70d84828485e4a726e10f9bf1
Author: Alexis Mousset <alexis.mousset@rudder.io>
Date:   Fri Jan 16 10:52:49 2026 +0100

    patch

diff --git a/configure b/configure
index 11cf67e..82cdb3e 100755
--- a/configure
+++ b/configure
@@ -16968,6 +16968,47 @@ _ACEOF
 fi
 
 
+# The cast to long int works around a bug in the HP C Compiler
+# version HP92453-01 B.11.11.23709.GP, which incorrectly rejects
+# declarations like 'int a3[[(sizeof (unsigned char)) >= 0]];'.
+# This bug is HP SR number 8606223364.
+{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking size of time_t" >&5
+printf %s "checking size of time_t... " >&6; }
+if test ${ac_cv_sizeof_time_t+y}
+then :
+  printf %s "(cached) " >&6
+else case e in #(
+  e) if ac_fn_c_compute_int "$LINENO" "(long int) (sizeof (time_t))" "ac_cv_sizeof_time_t"        "$ac_includes_default"
+then :
+
+else case e in #(
+  e) if test "$ac_cv_type_time_t" = yes; then
+     { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: error: in '$ac_pwd':" >&5
+printf "%s\n" "$as_me: error: in '$ac_pwd':" >&2;}
+as_fn_error 77 "cannot compute sizeof (time_t)
+See 'config.log' for more details" "$LINENO" 5; }
+   else
+     ac_cv_sizeof_time_t=0
+   fi ;;
+esac
+fi
+ ;;
+esac
+fi
+{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $ac_cv_sizeof_time_t" >&5
+printf "%s\n" "$ac_cv_sizeof_time_t" >&6; }
+
+
+
+printf "%s\n" "#define SIZEOF_TIME_T $ac_cv_sizeof_time_t" >>confdefs.h
+
+
+if test "x$ac_cv_sizeof_time_t" = "x8"; then
+
+printf "%s\n" "#define HAVE_64BIT_TIME_T 1" >>confdefs.h
+
+fi
+
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for an ANSI C-conforming const" >&5
 $as_echo_n "checking for an ANSI C-conforming const... " >&6; }
diff --git a/libpromises/cf3.defs.h b/libpromises/cf3.defs.h
index 6294c92..10ce3a9 100644
--- a/libpromises/cf3.defs.h
+++ b/libpromises/cf3.defs.h
@@ -556,7 +556,12 @@ typedef enum
 
 #define CF_BOOL      "true,false,yes,no,on,off"
 #define CF_LINKRANGE "symlink,hardlink,relative,absolute"
-#define CF_TIMERANGE "0,2147483647" /* i.e. "0,0x7fffffff" */
+
+#ifdef HAVE_64BIT_TIME_T
+# define CF_TIMERANGE "0,99999999999" /* Same as other max int values */
+#else
+# define CF_TIMERANGE "0,2147483647" /* i.e. "0,0x7fffffff" */
+#endif
 
 /* Syntax checker accepts absurdly big numbers for backwards
  * compatibility. WARNING: internally they are stored as longs, possibly
diff --git a/tests/acceptance/00_basics/03_bodies/file-select-Y2038.cf b/tests/acceptance/00_basics/03_bodies/file-select-Y2038.cf
new file mode 100644
index 0000000..69fc0db
--- /dev/null
+++ b/tests/acceptance/00_basics/03_bodies/file-select-Y2038.cf
@@ -0,0 +1,34 @@
+body common control
+{
+      inputs => {
+                  "../../default.cf.sub",
+      };
+      bundlesequence  => { default("$(this.promise_filename)") };
+      version => "1.0";
+}
+
+bundle agent init
+{
+  commands:
+      # Ensure there is a file with a date in the future beyond 2038
+      "/usr/bin/touch -t 204201010000 $(this.promise_filename).sub.file-with-mtime-after-2038";
+}
+
+bundle agent test
+{
+  meta:
+      "description" string => "Test that the agent does not crash on dates after 2038.";
+
+}
+
+bundle agent check
+{
+  meta:
+      "test_skip_unsupported" string => "!64_bit";
+
+  classes:
+      "ok" expression => "class_defined_from_included_bundle";
+  methods:
+      "" usebundle => dcs_passif_output(".*Pass.*", "", "$(sys.cf_agent) -Kf $(this.promise_filename).sub", $(this.promise_filename));
+
+}
diff --git a/tests/acceptance/00_basics/03_bodies/file-select-Y2038.cf.sub b/tests/acceptance/00_basics/03_bodies/file-select-Y2038.cf.sub
new file mode 100644
index 0000000..dbe58ef
--- /dev/null
+++ b/tests/acceptance/00_basics/03_bodies/file-select-Y2038.cf.sub
@@ -0,0 +1,32 @@
+bundle agent main {
+  files:
+
+  "$(this.promise_filename).sub.file-with-mtime-after-2038"
+    file_select => y2042,
+    perms => m( "o+r" ),
+    classes => kept_or_repaired("y2038_ok");
+
+  reports:
+    y2038_ok::
+      "Pass";
+}
+
+body file_select y2042 {
+ # January 1, 2042 at 12:00:00 AM December 31, 2042 at 11:59:59 PM
+ mtime => irange(2272896000, 2304431999);
+ file_result => "mtime.leaf_name";
+}
+
+body perms m (x)
+{
+  mode => "$(x)";
+}
+
+body classes kept_or_repaired(x)
+{
+  promise_repaired => { "$(x)" };
+  promise_kept => { "$(x)" };
+  repair_failed => { "NOT_$(x)" };
+  repair_denied => { "NOT_$(x)" };
+  repair_timeout => { "NOT_$(x)" };
+}
