diff --git a/libutils/logging.c b/libutils/logging.c
index ee488ed9b..c2d387c19 100644
--- a/libutils/logging.c
+++ b/libutils/logging.c
@@ -154,7 +154,7 @@ bool LoggingFormatTimestamp(char dest[64], size_t n, struct tm *timestamp)
 
 static void LogToConsole(const char *msg, LogLevel level, bool color)
 {
-    FILE *output_file = stdout; // Messages should ALL go to stdout else they are disordered
+    FILE *output_file = (level <= LOG_LEVEL_WARNING) ? stderr : stdout;
     struct tm now;
     time_t now_seconds = time(NULL);
     localtime_r(&now_seconds, &now);
@@ -165,16 +165,16 @@ static void LogToConsole(const char *msg, LogLevel level, bool color)
     }
     if (level >= LOG_LEVEL_INFO && VPREFIX[0])
     {
-        fprintf(stdout, "%s ", VPREFIX);
+        fprintf(output_file, "%s ", VPREFIX);
     }
     if (TIMESTAMPS)
     {
         char formatted_timestamp[64];
         LoggingFormatTimestamp(formatted_timestamp, 64, &now);
-        fprintf(stdout, "%s ", formatted_timestamp);
+        fprintf(output_file, "%s ", formatted_timestamp);
     }
 
-    fprintf(stdout, "%8s: %s\n", LogLevelToString(level), msg);
+    fprintf(output_file, "%8s: %s\n", LogLevelToString(level), msg);
 
     if (color)
     {
