--- debian/control	2015-02-04 16:54:57.379101905 +0100
+++ debian/control	2015-02-04 16:39:07.095356589 +0100
@@ -2,7 +2,7 @@
 Section: admin
 Priority: extra
 Maintainer: Rudder packaging team <rudder-packaging@rudder-project.org>
-Build-Depends: debhelper (>= 7), libssl-dev, bison, gcc, flex, autoconf, automake, libtool, libpcre3-dev, libpam0g-dev
+Build-Depends: debhelper (>= 7), bison, gcc, flex, autoconf, automake, libtool, libpcre3-dev, libpam0g-dev
 Standards-Version: 3.8.0
 Homepage: http://www.rudder-project.org
 
--- debian/rules	2015-02-04 16:54:57.379101905 +0100
+++ debian/rules	2015-02-04 16:39:13.271381445 +0100
@@ -17,6 +17,10 @@
 	dh_prep
 	# Add here commands to configure the package.
 	cd SOURCES && ./perl-prepare.sh $(CURDIR)/SOURCES/fusioninventory-agent
+	# Compile and install OpenSSL
+	cd SOURCES/openssl-source && if [ "$(shell dpkg-architecture -qDEB_HOST_ARCH)" = "amd64" ]; then TARGET=linux-x86_64; else TARGET=linux-elf; fi && ./Configure -fPIC --prefix=/opt/rudder --openssldir=/opt/rudder/openssl shared $$TARGET
+	cd SOURCES/openssl-source && make
+	cd SOURCES/openssl-source && make install INSTALL_PREFIX=$(CURDIR)/debian/tmp
 	# Compile the LMDB library and install it in /opt/rudder
 	# LMDB source code does not know how to create destination folders, do it ourselves
 	for i in bin lib include man/man1; do mkdir -p $(CURDIR)/debian/tmp/opt/rudder/$$i; done
@@ -25,7 +29,7 @@
 	# If there is no configure, bootstrap with autogen.sh first
 	cd SOURCES/cfengine-source && [ -x ./configure ] || NO_CONFIGURE=1 ./autogen.sh
 	# Prepare CFEngine 3.6.x build with defined path of LMDB
-	cd SOURCES/cfengine-source && ./configure --prefix=/opt/rudder --with-workdir=/var/rudder/cfengine-community --enable-static=yes --enable-shared=no --with-lmdb=$(CURDIR)/debian/tmp/opt/rudder CFLAGS="$(BUILD_CFLAGS) $(SECURE_CFLAGS)" LDFLAGS="$(BUILD_LDFLAGS) $(SECURE_LDFLAGS)"
+	cd SOURCES/cfengine-source && ./configure --prefix=/opt/rudder --with-workdir=/var/rudder/cfengine-community --enable-static=yes --enable-shared=no --with-openssl=$(CURDIR)/debian/tmp/opt/rudder --with-lmdb=$(CURDIR)/debian/tmp/opt/rudder CFLAGS="$(BUILD_CFLAGS) $(SECURE_CFLAGS)" LDFLAGS="$(BUILD_LDFLAGS) $(SECURE_LDFLAGS)"
 
 	touch configure-stamp
 
@@ -94,7 +98,7 @@
 	# Install an empty uuid.hive file before generating an uuid
 	dh_install --SOURCEDIR=$(CURDIR)/SOURCES/ uuid.hive /opt/rudder/etc/
 	# Install /etc/ld.so.conf.d/rudder.conf in order to use libraries contain
-	# in /opt/rudder/lib like LMDB
+	# in /opt/rudder/lib like LMDB and OpenSSL
 	dh_install --SOURCEDIR=$(CURDIR)/SOURCES/ rudder.conf /etc/ld.so.conf.d
 	# Install a verification script for cron
 	dh_install --SOURCEDIR=$(CURDIR)/SOURCES/ check-rudder-agent /opt/rudder/bin/
