--- a/source/src/generic_agent.c	(revision 1845)
+++ b/source/src/generic_agent.c	(revision 1846)
@@ -39,6 +39,17 @@
 
 /*****************************************************************************/
 
+static void SanitizeEnvironment()
+{
+    /* ps(1) and other utilities invoked by Cfengine may be affected */
+    unsetenv("COLUMNS");
+
+    /* Make sure subprocesses output is not localized */
+    unsetenv("LANG");
+    unsetenv("LANGUAGE");
+    unsetenv("LC_MESSAGES");
+}
+
 void GenericInitialize(int argc,char **argv,char *agents)
 
 { enum cfagenttype ag = Agent2Type(agents);
@@ -58,6 +69,7 @@
 SetReferenceTime(true);
 SetStartTime(false);
 SetSignals();
+SanitizeEnvironment();
 
 strcpy(THIS_AGENT,CF_AGENTTYPES[ag]);
 NewClass(THIS_AGENT);
