--- jetty/bin/jetty.sh.orig
+++ jetty/bin/jetty.sh
@@ -222,9 +222,23 @@ readConfig()
   source "$1"
 }

+# Gets the memory size (Xmx+MaxPermSize) needed by Java megabytes as argument.
+# Checks if there is enough available RAM + Swap to contain the JVM.
+checkAvailableRam()
+{
+  # By default, add 10% to the given needed memory size to have a safe
+  # margin (leave some memory for the OS)
+  TOTAL_MEM_NEEDED=$((((${1})*100)/90))
+  TOTAL_MEM_AVAILABLE=$(($(free -m|awk '/^Mem:/{print $2}')+$(free -m|awk '/^Swap:/{print $2}')))
+  if [ ${TOTAL_MEM_AVAILABLE} -le ${TOTAL_MEM_NEEDED} ]; then
+    echo "WARNING: Not enough free memory to start Jetty (about ${TOTAL_MEM_NEEDED}MB are needed). Trying anyway, but>
+  fi
+}
+
 dumpEnv()
 {
   echo "JAVA                  =  $JAVA"
+  echo "JAVA_XMX              =  ${JAVA_XMX}"
   echo "JAVA_OPTIONS          =  ${JAVA_OPTIONS[*]}"
   echo "JETTY_HOME            =  $JETTY_HOME"
   echo "JETTY_BASE            =  $JETTY_BASE"
@@ -439,6 +453,12 @@ then
   done < "$JETTY_CONF"
 fi

+##################################################
+# Set default JVM parameters, to be
+# overridden by the configuration file
+##################################################
+JAVA_XMX=${JAVA_XMX:-1024}
+
 ##################################################
 # Setup JAVA if unset
 ##################################################
@@ -549,6 +569,9 @@ case "$ACTION" in
     UMASK="0007"
     echo "Setting umask to ${UMASK}"

+    # Checking if enough RAM is available for Jetty to use
+    # Metaspace is auomanaged in JDK8+. Rudder needs 150Mo of it.
+    checkAvailableRam $((${JAVA_XMX}+150))

     # Startup from a service file
     if [ $UID -eq 0 ] && type start-stop-daemon > /dev/null 2>&1

