diff --git jetty/bin/jetty.sh jetty/bin/jetty.sh
index ebcf0f4..90707f6 100755
--- jetty/bin/jetty.sh
+++ jetty/bin/jetty.sh
@@ -307,6 +307,7 @@ fi
 #####################################################
 # define start log location
 #####################################################
+JETTY_START_LOG="/var/log/rudder/webapp/java.log"
 if [ -z "$JETTY_START_LOG" ]
 then
   JETTY_START_LOG="$JETTY_RUN/$NAME-start.log"
@@ -495,11 +496,11 @@ case "$ACTION" in
         # FIXME: Broken solution: wordsplitting, pathname expansion, arbitrary command execution, etc.
         su - "$JETTY_USER" $SU_SHELL -c "
           cd \"$JETTY_BASE\"
-          echo ${RUN_ARGS[*]} start-log-file=\"$JETTY_START_LOG\" | xargs ${JAVA} > /dev/null &
+          echo ${RUN_ARGS[*]} start-log-file=\"$JETTY_START_LOG\" | xargs ${JAVA} >${JETTY_START_LOG} 2>&1 &
           disown \$!
           echo \$! > \"$JETTY_PID\""
       else
-        echo ${RUN_ARGS[*]} | xargs ${JAVA} > /dev/null &
+        echo ${RUN_ARGS[*]} | xargs ${JAVA} >${JETTY_START_LOG} 2>&1 &
         disown $!
         echo $! > "$JETTY_PID"
       fi
