--- jetty/bin/jetty.sh.orig	2018-08-29 09:38:33.684462836 +0200
+++ jetty/bin/jetty.sh	2018-08-29 09:40:57.763691920 +0200
@@ -411,6 +411,7 @@ fi
 #####################################################
 # define start log location
 #####################################################
+JETTY_START_LOG="/var/log/rudder/webapp/java.log"
 if [ -z "$JETTY_START_LOG" ]
 then
   JETTY_START_LOG="$JETTY_RUN/$NAME-start.log"
@@ -621,13 +622,13 @@ case "$ACTION" in
         chown "$JETTY_USER" "$JETTY_PID"
         su - "$JETTY_USER" $SU_SHELL -c "
           cd \"$JETTY_BASE\"
-          echo ${RUN_ARGS[*]} --start-log-file=\"$JETTY_START_LOG\" | xargs ${JAVA} > /dev/null &
+          echo ${RUN_ARGS[*]} --start-log-file=\"$JETTY_START_LOG\" | xargs ${JAVA} >${JETTY_START_LOG} 2>&1 &
           PID=\$!
           disown \$PID"
         (( DEBUG )) && echo "Starting: su shell (w/user $JETTY_USER) on PID $PID"
       else
         # Startup if not switching users
-        echo ${RUN_ARGS[*]} --start-log-file="${JETTY_START_LOG}" | xargs ${JAVA} > /dev/null &
+        echo ${RUN_ARGS[*]} --start-log-file="${JETTY_START_LOG}" | xargs ${JAVA} >${JETTY_START_LOG} 2>&1 &
         PID=$!
         disown $PID
         (( DEBUG )) && echo "Starting: java command on PID $PID"

