--- jetty/bin/jetty.sh
+++ jetty/bin/jetty.sh
@@ -434,7 +434,7 @@ case "`uname`" in
 CYGWIN*) JETTY_START="`cygpath -w $JETTY_START`";;
 esac

-RUN_ARGS=$("$JAVA" -jar "$JETTY_START" --dry-run=opts,path,main,args ${JETTY_ARGS[*]})
+RUN_ARGS=$("$JAVA" -jar "$JETTY_START" --dry-run=opts,path,main,args ${JETTY_ARGS[*]} ${JAVA_OPTIONS[*]})
 RUN_CMD=("$JAVA" $JETTY_SYS_PROPS ${RUN_ARGS[@]})

 #####################################################
