--- jetty/bin/jetty.sh.orig
+++ jetty/bin/jetty.sh
@@ -451,6 +451,9 @@ fi
 case "$ACTION" in
   start)
     echo -n "Starting Jetty: "
+    UMASK="0007"
+    echo "Setting umask to ${UMASK}"
+

     if (( NO_START )); then
       echo "Not starting ${NAME} - NO_START=1";
@@ -465,7 +468,7 @@ case "$ACTION" in
         CH_USER="--chuid $JETTY_USER"
       fi

-      start-stop-daemon --start $CH_USER \
+      start-stop-daemon -k ${UMASK} --start $CH_USER \
        --pidfile "$JETTY_PID" \
        --chdir "$JETTY_BASE" \
        --background \

