--- jetty/bin/jetty.sh.orig	2018-08-28 17:55:24.439012644 +0200
+++ jetty/bin/jetty.sh	2018-08-28 17:58:28.054678897 +0200
@@ -154,9 +154,23 @@
   source "$1"
 }
 
+# Gets the memory size (Xmx+MaxPermSize) needed by Java megabytes as argument.
+# Checks if there is enough available RAM + Swap to contain the JVM.
+checkAvailableRam()
+{
+  # By default, add 10% to the given needed memory size to have a safe
+  # margin (leave some memory for the OS)
+  TOTAL_MEM_NEEDED=$((((${1})*100)/90))
+  TOTAL_MEM_AVAILABLE=$(($(free -m|awk '/^Mem:/{print $2}')+$(free -m|awk '/^Swap:/{print $2}')))
+  if [ ${TOTAL_MEM_AVAILABLE} -le ${TOTAL_MEM_NEEDED} ]; then
+    echo "WARNING: Not enough free memory to start Jetty (about ${TOTAL_MEM_NEEDED}MB are needed). Trying anyway, but the application is likely to fail."
+  fi
+}
+
 dumpEnv()
 {
     echo "JAVA                  =  $JAVA"
+    echo "JAVA_XMX              =  ${JAVA_XMX}"
     echo "JAVA_OPTIONS          =  ${JAVA_OPTIONS[*]}"
     echo "JETTY_HOME            =  $JETTY_HOME"
     echo "JETTY_BASE            =  $JETTY_BASE"
@@ -363,6 +377,12 @@
   done < "$JETTY_CONF"
 fi
 
+##################################################
+# Set default JVM parameters, to be
+# overridden by the configuration file
+##################################################
+JAVA_XMX=${JAVA_XMX:-1024}
+
 ##################################################
 # Setup JAVA if unset
 ##################################################
@@ -454,6 +474,9 @@
     UMASK="0007"
     echo "Setting umask to ${UMASK}"

+    # Checking if enough RAM is available for Jetty to use
+    # Metaspace is auomanaged in JDK8+. Rudder needs 150Mo of it.
+    checkAvailableRam $((${JAVA_XMX}+150))

     if (( NO_START )); then
       echo "Not starting ${NAME} - NO_START=1";
