--- a/lib/FusionInventory/Agent/Task/Inventory/Generic.pm
+++ b/lib/FusionInventory/Agent/Task/Inventory/Generic.pm
@@ -5,6 +5,7 @@
 
 use English qw(-no_match_vars);
 use Net::Domain qw(hostfqdn hostdomain);
+use FusionInventory::Agent::Tools;
 
 sub isEnabled {
     return 1;
@@ -15,9 +16,18 @@
 
     my $inventory = $params{inventory};
 
+    my $FQDN = hostfqdn();
+    # Sometimes hostfqdn is empty so we replace it with hostname
+    if ($FQDN eq "") {
+      my $command = $OSNAME eq 'linux' ? 'hostname --fqdn' : 'hostname';
+      $FQDN = getFirstLine(
+        command => $command
+      );
+    }
+
     $inventory->setOperatingSystem({
             KERNEL_NAME => $OSNAME,
-            FQDN => hostfqdn(),
+            FQDN => $FQDN,
             DNS_DOMAIN => hostdomain()
     });
 }
