--- fusioninventory-agent-2.3.6/lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Slackware.pm	2016-03-03 11:52:58.309514141 +0100
+++ fusioninventory-agent-2.3.6/lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Slackware.pm	2016-03-03 11:53:32.435513507 +0100
@@ -21,7 +21,7 @@
     return unless $handle;
 
     while (my $file = readdir($handle)) {
-        next unless $file =~ /^(.+)([^-]+-[^-]+-[^-]+)$/;
+        next unless $file =~ /^(.+)-(.+)-(i[0-9]86|noarch|x86_64|x86|fw|npmjs)-(.*)$/;
         my $name = $1;
         my $version = $2;
 
