diff -upr fusioninventory-agent/lib/FusionInventory/Agent/Task/Inventory/Generic/Users.pm fusioninventory-agent-b/lib/FusionInventory/Agent/Task/Inventory/Generic/Users.pm
--- fusioninventory-agent/lib/FusionInventory/Agent/Task/Inventory/Generic/Users.pm	2014-03-10 12:34:32.000000000 +0100
+++ fusioninventory-agent-b/lib/FusionInventory/Agent/Task/Inventory/Generic/Users.pm	2016-05-09 15:34:36.706050533 +0200
@@ -151,15 +151,25 @@ sub _getLastUser {
         @_
     );
 
-    my $last = getFirstLine(%params);
-    return unless $last;
-    return unless
-        $last =~ /^(\S+) \s+ \S+ \s+ \S+ \s+ (\S+ \s+ \S+ \s+ \S+ \s+ \S+)/x;
-
-    return {
-        LASTLOGGEDUSER     => $1,
-        DATELASTLOGGEDUSER => $2
-    };
+    my $handle = getFileHandle(%params);
+    return unless $handle;
+
+    my $last;
+
+    while (my $line = <$handle>) {
+        next if $line =~ /^reboot.*system boot/;
+        next unless $line =~ /^(\S+) \s+ \S+ \s+ \S+ \s+ (\S+ \s+ \S+ \s+ \S+ \s+ \S+)/x;
+
+        $last = {
+            LASTLOGGEDUSER     => $1,
+            DATELASTLOGGEDUSER => $2
+        };
+
+        last;
+    }
+    close $handle;
+
+    return $last;
 }
 
 1;

