--- fusioninventory-agent/lib/FusionInventory/Agent/Task/Inventory/Generic/Users.pm.orig    2017-02-06 10:35:26.000000000 +0100
+++ fusioninventory-agent/lib/FusionInventory/Agent/Task/Inventory/Generic/Users.pm 2017-02-06 10:36:00.000000000 +0100
@@ -157,6 +157,7 @@

     while (my $line = <$handle>) {
         next if $line =~ /^reboot.*system boot/;
+        next if $line =~ /^wtmp begins/;
         next unless $line =~ /^(\S+) \s+ \S+ \s+ \S+ \s+ (\S+ \s+ \S+ \s+ \S+ \s+ \S+)/x;

         $last = {
