diff --git a/lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/LSB.pm b/lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/LSB.pm
index 44d26ec..4415a37 100644
--- a/lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/LSB.pm
+++ b/lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/LSB.pm
@@ -5,8 +5,11 @@ use warnings;
 
 use FusionInventory::Agent::Tools;
 
+our $runMeIfTheseChecksFailed =
+    ["FusionInventory::Agent::Task::Inventory::Linux::Distro::OSRelease"];
+
 sub isEnabled {
-    return ! -r '/etc/SuSE-release' && ! -r '/etc/oracle-release' && canRun('lsb_release');
+    return ! -r '/etc/SuSE-release' && ! -r '/etc/oracle-release' && ! -r '/etc/os-release' && canRun('lsb_release');
 }
 
 sub doInventory {
diff --git a/lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/NonLSB.pm b/lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/NonLSB.pm
index dedfd53..336842c 100644
--- a/lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/NonLSB.pm
+++ b/lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/NonLSB.pm
@@ -66,7 +66,7 @@ our $runMeIfTheseChecksFailed =
     ["FusionInventory::Agent::Task::Inventory::Linux::Distro::LSB"];
 
 sub isEnabled {
-    return -r '/etc/SuSE-release' || -r '/etc/oracle-release' || !canRun('lsb_release');
+    return -r '/etc/SuSE-release' || -r '/etc/oracle-release' || !canRun('lsb_release') || ! -r '/etc/os-release';
 }
 
 sub doInventory {
diff --git a/lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/OSRelease.pm b/lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/OSRelease.pm
new file mode 100644
index 0000000..2e77b9f
--- /dev/null
+++ b/lib/FusionInventory/Agent/Task/Inventory/Linux/Distro/OSRelease.pm
@@ -0,0 +1,43 @@
+package FusionInventory::Agent::Task::Inventory::Linux::Distro::OSRelease;
+
+use strict;
+use warnings;
+
+use FusionInventory::Agent::Tools;
+
+sub isEnabled {
+  return -r '/etc/os-release';
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    my $handle = getFileHandle(
+        logger  => $logger,
+        file => '/etc/os-release',
+    );
+
+    my ($name, $version, $description);
+    while (my $line = <$handle>) {
+        $name        = $1 if $line =~ /^NAME="?([^"]+)"?/;
+        $version     = $1 if $line =~ /^VERSION="?([^"]+)"?/;
+        $description = $1 if $line =~ /^PRETTY_NAME="?([^"]+)"?/;
+    }
+    close $handle;
+
+    $inventory->setHardware({
+        OSNAME => $description,
+    });
+
+    $inventory->setOperatingSystem({
+        NAME      => $name,
+        VERSION   => $version,
+        FULL_NAME => $description
+    });
+
+}
+
+1;

