diff --color -ruN fusion-source.bak/lib/FusionInventory/Agent/Task/Inventory/Linux/Drives.pm fusion-source/lib/FusionInventory/Agent/Task/Inventory/Linux/Drives.pm
--- fusion-source.bak/lib/FusionInventory/Agent/Task/Inventory/Linux/Drives.pm	2020-11-09 15:29:50.000000000 +0100
+++ fusion-source/lib/FusionInventory/Agent/Task/Inventory/Linux/Drives.pm	2023-05-23 10:54:06.561162476 +0200
@@ -33,12 +33,20 @@
 sub _getFilesystems {
     my ($logger) = @_;
 
+    # if timeout is available, use it for df command to avoid hanging on nfs issues
+    my $df_command;
+    if (canRun('timeout')) {
+      $df_command = 'timeout 10 df -P -T -k';
+    } else {
+      $df_command = 'df -P -T -k';
+    }
+
     # get filesystems list
     my @filesystems =
         # exclude virtual file systems and overlay fs defined by docker
         grep { $_->{FILESYSTEM} !~ /^(tmpfs|devtmpfs|usbfs|proc|devpts|devshm|udev)$/ && $_->{VOLUMN} !~ /^overlay$/ }
         # get all file systems
-        getFilesystemsFromDf(logger => $logger, command => 'df -P -T -k');
+        getFilesystemsFromDf(logger => $logger, command => $df_command);
 
     # get additional informations
     if (canRun('blkid')) {
