diff --color -ruN fusion-source.bak/lib/FusionInventory/Agent/Inventory.pm fusion-source/lib/FusionInventory/Agent/Inventory.pm
--- fusion-source.bak/lib/FusionInventory/Agent/Inventory.pm	2023-05-23 11:01:21.586280874 +0200
+++ fusion-source/lib/FusionInventory/Agent/Inventory.pm	2023-05-23 11:04:24.667032086 +0200
@@ -87,7 +87,7 @@
                             NAME NO_REMOVE RELEASE_TYPE PUBLISHER
                             UNINSTALL_STRING URL_INFO_ABOUT VERSION
                             VERSION_MINOR VERSION_MAJOR GUID ARCH USERNAME
-                            USERID SYSTEM_CATEGORY/ ],
+                            USERID SYSTEM_CATEGORY SOURCE_NAME SOURCE_VERSION/ ],
     SOUNDS           => [ qw/CAPTION DESCRIPTION MANUFACTURER NAME/ ],
     STORAGES         => [ qw/DESCRIPTION DISKSIZE INTERFACE MANUFACTURER MODEL
                             NAME TYPE SERIAL SERIALNUMBER FIRMWARE SCSI_COID
diff --color -ruN fusion-source.bak/lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Deb.pm fusion-source/lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Deb.pm
--- fusion-source.bak/lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Deb.pm	2020-08-04 09:36:42.000000000 +0200
+++ fusion-source/lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Deb.pm	2023-05-23 11:02:52.880603498 +0200
@@ -24,7 +24,9 @@
         '${Version}\t'.
         '${Installed-Size}\t'.
         '${Section}\t' .
-        '${Status}\n' .
+        '${Status}\t' .
+        '${Source}\t' .
+        '${source:Version}\n' .
         '\'';
 
     my $packages = _getPackagesList(
@@ -75,6 +77,8 @@
             ARCH        => $infos[1],
             VERSION     => $infos[2],
             FILESIZE    => $infos[3] =~ /^\d+$/ ? $infos[3]*1024 : 0,
+            SOURCE_NAME => $infos[6],
+            SOURCE_VERSION => $infos[7],
             FROM        => 'deb',
             SYSTEM_CATEGORY => $infos[4]
         };
