--- fusioninventory-agent/lib/FusionInventory/Agent/Task/Inventory/Virtualization/Vmsystem.pm
+++ fusioninventory-agent/lib/FusionInventory/Agent/Task/Inventory/Virtualization/Vmsystem.pm
@@ -97,6 +97,7 @@ sub _getType {
     my ($bios, $logger) = @_;
 
     if ($bios->{SMANUFACTURER}) {
+        return 'QEMU'    if $bios->{SMANUFACTURER} =~ /QEMU/;
         return 'Hyper-V' if $bios->{SMANUFACTURER} =~ /Microsoft/;
         return 'VMware'  if $bios->{SMANUFACTURER} =~ /VMware/;
     }
