--- jetty/bin/jetty.sh.orig	2018-08-29 09:38:33.684462836 +0200
+++ jetty/bin/jetty.sh	2018-08-29 09:40:57.763691920 +0200
@@ -317,6 +317,7 @@
 #####################################################
 # define start log location
 #####################################################
+JETTY_START_LOG="/var/log/rudder/webapp/java.log"
 if [ -z "$JETTY_START_LOG" ]
 then
   JETTY_START_LOG="$JETTY_RUN/$NAME-start.log"
@@ -520,11 +521,11 @@
         # FIXME: Broken solution: wordsplitting, pathname expansion, arbitrary command execution, etc.
         su - "$JETTY_USER" $SU_SHELL -c "
           cd \"$JETTY_BASE\"
-          exec ${RUN_CMD[*]} start-log-file=\"$JETTY_START_LOG\" > /dev/null &
+          exec ${RUN_CMD[*]} start-log-file=\"$JETTY_START_LOG\" >${JETTY_START_LOG} 2>&1 &
           disown \$!
           echo \$! > \"$JETTY_PID\""
       else
-        "${RUN_CMD[@]}" > /dev/null &
+        "${RUN_CMD[@]}" >${JETTY_START_LOG} 2>&1 &
         disown $!
         echo $! > "$JETTY_PID"
       fi
