diff -Naur ../cfengine-3.5.2/cf-agent/nfs.c ./cf-agent/nfs.c
--- ../cfengine-3.5.2/cf-agent/nfs.c	2013-08-30 13:01:29.000000000 +0200
+++ ./cf-agent/nfs.c	2014-02-25 08:57:17.470905000 +0100
@@ -51,6 +51,7 @@
 static const char *VMOUNTCOMM[PLATFORM_CONTEXT_MAX] =
 {
     "",
+    "/bin/mount -va",           /* virt_host_vz_vzps */
     "/sbin/mount -ea",          /* hpux */
     "/usr/sbin/mount -t nfs",   /* aix */
     "/bin/mount -va",           /* linux */
@@ -72,6 +73,7 @@
 static const char *VUNMOUNTCOMM[PLATFORM_CONTEXT_MAX] =
 {
     "",
+    "/bin/umount",              /* virt_host_vz_vzps */
     "/sbin/umount",             /* hpux */
     "/usr/sbin/umount",         /* aix */
     "/bin/umount",              /* linux */
@@ -93,6 +95,7 @@
 static const char *VMOUNTOPTS[PLATFORM_CONTEXT_MAX] =
 {
     "",
+    "defaults",                 /* virt_host_vz_vzps */
     "bg,hard,intr",             /* hpux */
     "bg,hard,intr",             /* aix */
     "defaults",                 /* linux */
diff -Naur ../cfengine-3.5.2/cf-monitord/mon_network.c ./cf-monitord/mon_network.c
--- ../cfengine-3.5.2/cf-monitord/mon_network.c	2013-08-30 13:01:29.000000000 +0200
+++ ./cf-monitord/mon_network.c	2014-02-25 08:57:17.470905000 +0100
@@ -75,6 +75,7 @@
 static const char *VNETSTAT[PLATFORM_CONTEXT_MAX] =
 {
     "-",
+    "/bin/netstat -rn",         /* virt_host_vz_vzps */
     "/usr/bin/netstat -rn",     /* hpux */
     "/usr/bin/netstat -rn",     /* aix */
     "/bin/netstat -rn",         /* linux */
diff -Naur ../cfengine-3.5.2/cf-monitord/mon_processes.c ./cf-monitord/mon_processes.c
--- ../cfengine-3.5.2/cf-monitord/mon_processes.c	2013-08-30 13:01:29.000000000 +0200
+++ ./cf-monitord/mon_processes.c	2014-02-25 11:47:08.758905001 +0100
@@ -77,7 +77,7 @@
     char user[CF_MAXVARSIZE];
     char vbuff[CF_BUFSIZE];
 
-    snprintf(pscomm, CF_BUFSIZE, "%s %s", VPSCOMM[VSYSTEMHARDCLASS], VPSOPTS[VSYSTEMHARDCLASS]);
+    snprintf(pscomm, CF_BUFSIZE, "%s %s", VPSCOMM[VPSHARDCLASS], VPSOPTS[VPSHARDCLASS]);
 
     if ((pp = cf_popen(pscomm, "r", true)) == NULL)
     {
diff -Naur ../cfengine-3.5.2/libpromises/cf3.defs.h ./libpromises/cf3.defs.h
--- ../cfengine-3.5.2/libpromises/cf3.defs.h	2013-08-30 13:01:29.000000000 +0200
+++ ./libpromises/cf3.defs.h	2014-02-25 11:42:21.414905000 +0100
@@ -233,6 +233,7 @@
 typedef enum
 {
     PLATFORM_CONTEXT_UNKNOWN,
+    PLATFORM_CONTEXT_VZ_VZPS,
     PLATFORM_CONTEXT_HP,
     PLATFORM_CONTEXT_AIX,
     PLATFORM_CONTEXT_LINUX,
diff -Naur ../cfengine-3.5.2/libpromises/cf3.extern.h ./libpromises/cf3.extern.h
--- ../cfengine-3.5.2/libpromises/cf3.extern.h	2013-08-30 13:01:29.000000000 +0200
+++ ./libpromises/cf3.extern.h	2014-02-25 11:44:10.550905002 +0100
@@ -59,6 +59,7 @@
 
 extern char VDOMAIN[CF_MAXVARSIZE];
 extern PlatformContext VSYSTEMHARDCLASS;
+extern PlatformContext VPSHARDCLASS; /* used to define which ps command to use*/
 extern char VFQNAME[];
 extern char VUQNAME[];
 
diff -Naur ../cfengine-3.5.2/libpromises/cf3globals.c ./libpromises/cf3globals.c
--- ../cfengine-3.5.2/libpromises/cf3globals.c	2013-08-30 13:01:29.000000000 +0200
+++ ./libpromises/cf3globals.c	2014-02-25 11:43:31.118905001 +0100
@@ -116,3 +116,4 @@
 bool MINUSF = false;
 
 PlatformContext VSYSTEMHARDCLASS;
+PlatformContext VPSHARDCLASS; /* used to define which ps command to use*/
diff -Naur ../cfengine-3.5.2/libpromises/classes.c ./libpromises/classes.c
--- ../cfengine-3.5.2/libpromises/classes.c	2013-08-30 13:01:29.000000000 +0200
+++ ./libpromises/classes.c	2014-02-25 08:57:17.470905000 +0100
@@ -27,6 +27,7 @@
 const char *CLASSTEXT[PLATFORM_CONTEXT_MAX] =
 {
     "<unknown>",
+    "virt_host_vz_vzps",
     "hpux",
     "aix",
     "linux",
@@ -48,6 +49,7 @@
 const char *VPSCOMM[PLATFORM_CONTEXT_MAX] =
 {
     "",
+    "/bin/vzps -E 0",           /* vz with vzps */
     "/bin/ps",                  /* hpux */
     "/bin/ps",                  /* aix */
     "/bin/ps",                  /* linux */
@@ -72,6 +74,7 @@
 const char *VPSOPTS[PLATFORM_CONTEXT_MAX] =
 {
     "",
+    "-eo user,pid,ppid,pgid,pcpu,pmem,vsz,pri,rss,thcount,stime,time,args", /* vz with vzps */
     "-ef",                      /* hpux */
     "-N -eo user,pid,ppid,pgid,pcpu,pmem,vsz,ni,stat,st=STIME,time,args",  /* aix */
     "-eo user,pid,ppid,pgid,pcpu,pmem,vsz,ni,rss,nlwp,stime,time,args",        /* linux */
@@ -93,6 +96,7 @@
 const char *VFSTAB[PLATFORM_CONTEXT_MAX] =
 {
     "-",
+    "/etc/fstab",               /* vz with vzps */
     "/etc/fstab",               /* hpux */
     "/etc/filesystems",         /* aix */
     "/etc/fstab",               /* linux */
diff -Naur ../cfengine-3.5.2/libpromises/processes_select.c ./libpromises/processes_select.c
--- ../cfengine-3.5.2/libpromises/processes_select.c	2013-08-30 13:01:29.000000000 +0200
+++ ./libpromises/processes_select.c	2014-02-25 11:48:40.442905001 +0100
@@ -655,7 +655,7 @@
 
     if (strcmp(zone, "global") == 0)
     {
-        snprintf(psopts, CF_BUFSIZE, "%s,zone", VPSOPTS[VSYSTEMHARDCLASS]);
+        snprintf(psopts, CF_BUFSIZE, "%s,zone", VPSOPTS[VPSHARDCLASS]);
         return psopts;
     }
 # endif
@@ -669,7 +669,7 @@
 
 # endif
 
-    return VPSOPTS[VSYSTEMHARDCLASS];
+    return VPSOPTS[VPSHARDCLASS];
 }
 #endif
 
@@ -774,7 +774,7 @@
 
     const char *psopts = GetProcessOptions();
 
-    snprintf(pscomm, CF_MAXLINKSIZE, "%s %s", VPSCOMM[VSYSTEMHARDCLASS], psopts);
+    snprintf(pscomm, CF_MAXLINKSIZE, "%s %s", VPSCOMM[VPSHARDCLASS], psopts);
 
     Log(LOG_LEVEL_VERBOSE, "Observe process table with %s", pscomm);
 
diff -Naur ../cfengine-3.5.2/libpromises/sysinfo.c ./libpromises/sysinfo.c
--- ../cfengine-3.5.2/libpromises/sysinfo.c	2013-08-30 13:01:29.000000000 +0200
+++ ./libpromises/sysinfo.c	2014-02-25 12:04:16.086905001 +0100
@@ -76,6 +76,7 @@
 static int Xen_Domain(EvalContext *ctx);
 static int EOS_Version(EvalContext *ctx);
 static int MiscOS(EvalContext *ctx);
+static void OpenVZ_Detect(EvalContext *ctx);
 
 #ifdef XEN_CPUID_SUPPORT
 static void Xen_Cpuid(uint32_t idx, uint32_t *eax, uint32_t *ebx, uint32_t *ecx, uint32_t *edx);
@@ -91,6 +92,7 @@
 static const char *CLASSATTRIBUTES[PLATFORM_CONTEXT_MAX][3] =
 {
     {"-", "-", "-"},            /* as appear here are matched. The fields are sysname and machine */
+    {"virt_host_vz_vzps", ".*", ".*"},      /* VZ Host with vzps installed (virt_host_vz_vzps) */
     {"hp-ux", ".*", ".*"},      /* hpux */
     {"aix", ".*", ".*"},        /* aix */
     {"linux", ".*", ".*"},      /* linux */
@@ -112,6 +114,7 @@
 static const char *VRESOLVCONF[PLATFORM_CONTEXT_MAX] =
 {
     "-",
+    "/etc/resolv.conf",         /* virt_host_vz_vzps */
     "/etc/resolv.conf",         /* hpux */
     "/etc/resolv.conf",         /* aix */
     "/etc/resolv.conf",         /* linux */
@@ -133,6 +136,7 @@
 static const char *VMAILDIR[PLATFORM_CONTEXT_MAX] =
 {
     "-",
+    "/var/spool/mail",          /* virt_host_vz_vzps */
     "/var/mail",                /* hpux */
     "/var/spool/mail",          /* aix */
     "/var/spool/mail",          /* linux */
@@ -154,6 +158,7 @@
 static const char *VEXPORTS[PLATFORM_CONTEXT_MAX] =
 {
     "-",
+    "/etc/exports",             /* virt_host_vz_vzps */
     "/etc/exports",             /* hpux */
     "/etc/exports",             /* aix */
     "/etc/exports",             /* linux */
@@ -359,6 +364,7 @@
                     found = true;
 
                     VSYSTEMHARDCLASS = (PlatformContext) i;
+                    VPSHARDCLASS = (PlatformContext) i; /* this one can be overriden at vz detection */
                     ScopeNewSpecial(ctx, "sys", "class", CLASSTEXT[i], DATA_TYPE_STRING);
                     break;
                 }
@@ -951,6 +957,11 @@
     {
         MiscOS(ctx);
     }
+
+    if (stat("/proc/self/status", &statbuf) != -1)
+    {
+        OpenVZ_Detect(ctx);
+    }
     
 #ifdef XEN_CPUID_SUPPORT
     else if (Xen_Hv_Check())
@@ -2205,6 +2216,52 @@
 }
 
 /******************************************************************/
+static void OpenVZ_Detect(EvalContext *ctx)
+{
+/* paths to file defining the type of vm (guest or host) */
+#define OPENVZ_HOST_FILENAME "/proc/bc/0"
+#define OPENVZ_GUEST_FILENAME "/proc/vz"
+/* path to the vzps binary */
+#define OPENVZ_VZPS_FILE "/bin/vzps"
+    struct stat statbuf;
+    int i;
+
+    /* The file /proc/bc/0 is present on host
+       The file /proc/vz is present on guest
+       If the host has /bin/vzps, we should use it for checking processes
+    */
+
+    if (stat(OPENVZ_HOST_FILENAME, &statbuf) != -1)
+    {
+        Log(LOG_LEVEL_VERBOSE, "This appears to be an OpenVZ/Virtuozzo/Parallels Cloud Server host system.\n");
+        EvalContextHeapAddHard(ctx, "virt_host_vz");
+        /* if the file /bin/vzps is there, it is safe to use the processes promise type */ 
+        if (stat(OPENVZ_VZPS_FILE, &statbuf) != -1)
+        {
+            EvalContextHeapAddHard(ctx, "virt_host_vz_vzps");
+            /* here we must redefine the value of VPSHARDCLASS */
+            for (i = 0; i < PLATFORM_CONTEXT_MAX; i++)
+            {
+                if (FullTextMatch(CLASSATTRIBUTES[i][0], "virt_host_vz_vzps"))
+                {
+                    VPSHARDCLASS = (PlatformContext) i;
+                    break;
+                }
+            }
+        }
+        else
+        {
+            Log(LOG_LEVEL_NOTICE, "This OpenVZ/Virtuozzo/Parallels Cloud Server host system does not have vzps installed; the processes promise type may not work as expected.\n");
+        }
+    }
+    else if (stat(OPENVZ_GUEST_FILENAME, &statbuf) != -1)
+    {
+        Log(LOG_LEVEL_VERBOSE, "This appears to be an OpenVZ/Virtuozzo/Parallels Cloud Server guest system.\n");
+        EvalContextHeapAddHard(ctx, "virt_guest_vz");
+    }
+}
+
+/******************************************************************/
 
 #ifdef XEN_CPUID_SUPPORT
 
