diff --git a/src/cf3.defs.h b/src/cf3.defs.h
index 3f44844..65d87b8 100755
--- a/src/cf3.defs.h
+++ b/src/cf3.defs.h
@@ -46,7 +46,7 @@
 /* Various defines                                                 */
 /*******************************************************************/
 
-#define CF_BUFSIZE 4096
+#define CF_BUFSIZE 8192
 /* max size of plaintext in one transaction, see
    net.c:SendTransaction(), leave space for encryption padding
    (assuming max 64*8 = 512-bit cipher block size)*/
