--- Makefile.PL.orig	2014-03-13 00:27:37.478644378 +0100
+++ Makefile.PL	2014-03-13 00:27:37.106645811 +0100
@@ -158,7 +158,8 @@
     $install .= <<'EOF';
 
 config_install :
-	install -d -m 755 $(DESTDIR)$(SYSCONFDIR)
+	mkdir -p $(DESTDIR)$(SYSCONFDIR)
+	chown 755 $(DESTDIR)$(SYSCONFDIR)
 	if [ -f $(DESTDIR)/$(SYSCONFDIR)/agent.cfg ]; then \
 	    install -m 644 etc/agent.cfg $(DESTDIR)$(SYSCONFDIR)/agent.cfg.new; \
 	else \
@@ -166,9 +167,9 @@
 	fi
 
 data_install :
-	install -d -m 755 $(DESTDIR)$(DATADIR)
+	mkdir -p $(DESTDIR)$(DATADIR) $(DESTDIR)$(DATADIR)/html
+	chown 755 $(DESTDIR)$(DATADIR) $(DESTDIR)$(DATADIR)/html
 	install -m 644 share/pci.ids $(DESTDIR)$(DATADIR)/
-	install -d -m 755 $(DESTDIR)$(DATADIR)/html
 	install -m 644 share/html/* $(DESTDIR)$(DATADIR)/html
 EOF
     return $install;
