diff --git a/rudder-agent/debian/control b/rudder-agent/debian/control
index 7b326dc9..ce7f4047 100644
--- a/control
+++ b/control
@@ -1,4 +1,4 @@
-Source: rudder-agent
+Source: rudder-agent-thin
 Section: admin
 Priority: extra
 Maintainer: Rudder packaging team <rudder-packaging@rudder-project.org>
@@ -6,14 +6,14 @@ Build-Depends: debhelper (>= 7), libssl-dev, bison, gcc, flex, autoconf, automak
 Standards-Version: 3.8.0
 Homepage: http://www.rudder-project.org
 
-Package: rudder-agent
+Package: rudder-agent-thin
 Architecture: any
-Depends: ${shlibs:Depends}, ${misc:Depends}, uuid-runtime, dmidecode, cron, net-tools ${rudder:deps}
+Depends: ${shlibs:Depends}, ${misc:Depends}, uuid-runtime, dmidecode, cron, net-tools, fusioninventory-agent ${rudder:deps}
 # The dependencies below are defined in order to use rudder-agent
 # for the server. This will add capabilities to send inventories
 # from the server itself.
 Provides: rudder-agent
-Conflicts: rudder-agent-thin
+Conflicts: rudder-agent
 Description: Configuration management and audit tool - agent
  Rudder is an open source configuration management and audit solution.
  .
