diff --git a/control b/control
index 15880ec..cbf4500 100644
--- a/control
+++ b/control
@@ -1,4 +1,4 @@
-Source: rudder-agent
+Source: rudder-agent-thin
 Section: admin
 Priority: extra
 Maintainer: Rudder packaging team <rudder-packaging@rudder-project.org>
@@ -6,19 +6,13 @@ Build-Depends: debhelper (>= 7), libssl-dev, bison, gcc, flex, autoconf, automak
 Standards-Version: 3.8.0
 Homepage: http://www.rudder-project.org
 
-Package: rudder-agent
+Package: rudder-agent-thin
 Architecture: any
-Depends: ${shlibs:Depends}, ${misc:Depends}, uuid-runtime, dmidecode, cron, net-tools, openssl
-# The dependencies below are defined in order to use rudder-agent
-# for the server. This will add capabilities to send inventories
-# from the server itself.
-Provides: rudder-cfengine-community, rudder-agent
-Conflicts: rudder-cfengine-community, rudder-agent-thin
-Replaces: rudder-cfengine-community, rudder-server-root (<< 2.4.0~beta2)
-Breaks: rudder-server-root (<< 2.4.0~beta2)
-Description: Configuration management and audit tool - agent
+Depends: ${shlibs:Depends}, ${misc:Depends}, uuid-runtime, dmidecode, cron, net-tools, openssl, fusioninventory-agent
+Provides: rudder-agent
+Conflicts: rudder-agent
+Description: Configuration management and audit tool
  Rudder is an open source configuration management and audit solution.
  .
  This package contains the agent that must be installed on all nodes to be
- managed by Rudder. It is based on two main components: CFEngine Community 3 and
- FusionInventory.
+ managed by Rudder. It is based on CFEngine Community.
-
