--- a/rudder-agent.spec	2014-12-08 15:05:33.466210433 +0100
+++ b/rudder-agent.spec	2014-12-08 15:10:56.454760369 +0100
@@ -17,7 +17,7 @@
 #####################################################################################
 
 #=================================================
-# Specification file for rudder-agent
+# Specification file for rudder-agent-thin
 #
 # Install CFEngine
 # Install Rudder initial promises
@@ -29,7 +29,7 @@
 #=================================================
 # Variables
 #=================================================
-%define real_name            rudder-agent
+%define real_name            rudder-agent-thin
 
 %define rudderdir            /opt/rudder
 %define ruddervardir         /var/rudder
@@ -69,7 +69,6 @@
 Source8: vzps.py
 Source9: rudder-agent.sh
 Source10: detect_os.sh
-Source11: rudder-perl
 Source12: rudder-agent-utilities
 
 # uuidgen doesn't exist on AIX, so we provide a simple shell compatible version
@@ -77,11 +76,6 @@
 Source100: uuidgen
 %endif
 
-# Prevent dependency auto-generation, that tries to be helpful by detecting Perl dependencies from
-# FusionInventory. We handle that with the perl standalone installation already.
-AutoReq: 0
-AutoProv: 0
-
 %if 0%{?rhel} && 0%{?rhel} == 4
 Patch1: fix-missing-headers
 %endif
@@ -89,10 +83,10 @@
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 
 # Generic requirements
-BuildRequires: gcc bison flex pcre-devel autoconf automake libtool
-Requires: pcre
+BuildRequires: gcc bison flex openssl-devel pcre-devel autoconf automake libtool
+Requires: openssl pcre fusioninventory-agent fusioninventory-agent-task-inventory
 Provides: rudder-agent
-Conflicts: rudder-agent-thin
+Conflicts: rudder-agent
 
 # Specific requirements
 
@@ -191,6 +185,10 @@
 %define use_system_openssl false
 %endif
 
+# rudder-agent-thin specific override: we ALWAYS want to use
+# the system OpenSSL version.
+%define use_system_openssl true
+
 ## 5 - Resulting dependencies
 %if "%{use_system_openssl}" == "true"
 BuildRequires: openssl-devel
@@ -207,23 +205,11 @@
 %define cp_a_command           cp -hpPr
 %endif
 
-# Replaces rudder-cfengine-community since 2.4.0~beta3
-Provides: rudder-cfengine-community
-Obsoletes: rudder-cfengine-community
-
-# Use our own dependency generator
-%global _use_internal_dependency_generator 0
-%global __find_requires_orig %{__find_requires}
-%define __find_requires %{_sourcedir}/filter-reqs.pl %{use_system_lmdb} %{__find_requires_orig}
-%global __find_provides_orig %{__find_provides}
-%define __find_provides %{_sourcedir}/filter-reqs.pl %{use_system_lmdb} %{__find_provides_orig}
-
 %description
 Rudder is an open source configuration management and audit solution.
 
 This package contains the agent that must be installed on all nodes to be
-managed by Rudder. It is based on two main components: CFEngine Community 3 and
-FusionInventory.
+managed by Rudder. It is based on CFEngine Community.
 
 #=================================================
 # Source preparation
@@ -240,8 +226,6 @@
 
 cd %{_sourcedir}
 
-%{_sourcedir}/perl-prepare.sh %{_sourcedir}/fusioninventory-agent
-
 # Ensure an appropriate environment for the compiler
 export CFLAGS="$RPM_OPT_FLAGS"
 export CXXFLAGS="$RPM_OPT_FLAGS"
@@ -365,9 +349,6 @@
 # Initial promises
 cp -r %{_sourcedir}/initial-promises %{buildroot}%{rudderdir}/share/
 
-# Fusion
-%{cp_a_command} %{_sourcedir}/perl-custom/opt/rudder/* %{buildroot}%{rudderdir}
-
 # Wrapper script
 %{install_command} -m 755 %{SOURCE3} %{buildroot}/opt/rudder/bin/run-inventory
 
@@ -390,8 +371,6 @@
 
 %{install_command} -m 755 %{SOURCE8} %{buildroot}/opt/rudder/bin/vzps.py
 
-%{install_command} -m 755 %{SOURCE11} %{buildroot}/opt/rudder/bin/rudder-perl
-
 # Rudder agent utilities
 %{install_command} -m 755 %{SOURCE12}/bin/rudder %{buildroot}%{rudderdir}/bin/rudder
 %{cp_a_command} %{SOURCE12}/share/commands/* %{buildroot}%{rudderdir}/share/commands/
@@ -416,7 +395,7 @@
 # Build a list of files to include in this package for use in the %files section below
 find %{buildroot}%{rudderdir} %{buildroot}%{ruddervardir} -type f -o -type l | sed "s,%{buildroot},," | sed "s,\.py$,\.py*," | grep -v "%{rudderdir}/etc/uuid.hive" | grep -v "%{ruddervardir}/cfengine-community/ppkeys" > %{_builddir}/file.list.%{name}
 
-%pre -n rudder-agent
+%pre -n rudder-agent-thin
 #=================================================
 # Pre Installation
 #=================================================
@@ -437,7 +416,7 @@
 %endif
 fi
 
-%post -n rudder-agent
+%post -n rudder-agent-thin
 #=================================================
 # Post Installation
 #=================================================
@@ -672,7 +651,7 @@
 
 /opt/rudder/bin/check-rudder-agent
 
-%preun -n rudder-agent
+%preun -n rudder-agent-thin
 #=================================================
 # Pre Uninstallation
 #=================================================
@@ -694,7 +673,7 @@
 fi
 
 
-%postun -n rudder-agent
+%postun -n rudder-agent-thin
 #=================================================
 # Post Uninstallation
 #=================================================
@@ -741,7 +720,7 @@
 # Files
 #=================================================
 # Files from %{rudderdir} and %{ruddervardir} are automatically added via the -f option
-%files -n rudder-agent -f %{_builddir}/file.list.%{name}
+%files -n rudder-agent-thin -f %{_builddir}/file.list.%{name}
 %defattr(-, root, root, 0755)
 
 %{bindir}/rudder
@@ -779,15 +758,8 @@
 # Changelog
 #=================================================
 %changelog
-* Wed Apr  27 2011 - Matthieu CERDA <matthieu.cerda@normation.com> 2.2-beta1-2
-- The packages now builds correctly on both x86 and x86_64 archs, and on EL4/CentOS 4.
-* Tue Mar  1 2011 - Jonathan CLARKE <jonathan.clarke@normation.com> 2.2-beta1-1
-- Release 2.2.beta1
-* Fri Feb 25 2011 - Jonathan CLARKE <jonathan.clarke@normation.com> 2.2-beta0-4
-- Fix bug in postinstall script - stop daemons before replacing them!
-* Fri Feb 25 2011 - Jonathan CLARKE <jonathan.clarke@normation.com> 2.2-beta0-3
-- Fix bug to get initial promises in RPM, using the right git branch
-* Fri Feb 25 2011 - Jonathan CLARKE <jonathan.clarke@normation.com> 2.2-beta0-2
-- Fix bug to get initial promises in RPM
-* Fri Feb 25 2011 - Jonathan CLARKE <jonathan.clarke@normation.com> 2.2-beta0-1
-- Initial package
+* Fri May  30 2014 - Matthieu CERDA <matthieu.cerda@normation.com> 2.11-beta1
+- Initial package, using rudder-agent as a base
+- Removed fusion-inventory code
+- Removed legacy code
+
