--- jetty7/bin/jetty.sh.orig	2014-03-13 01:12:07.659327820 +0100
+++ jetty7/bin/jetty.sh	2014-03-13 01:10:00.507714504 +0100
@@ -134,6 +134,18 @@
   source "$1"
 }
 
+# Gets the memory size (Xmx+MaxPermSize) needed by Java megabytes as argument.
+# Checks if there is enough available RAM + Swap to contain the JVM.
+checkAvailableRam()
+{
+  # By default, add 10% to the given needed memory size to have a safe
+  # margin (leave some memory for the OS)
+  TOTAL_MEM_NEEDED=$((((${1})*100)/90))
+  TOTAL_MEM_AVAILABLE=$(($(free -m|awk '/^Mem:/{print $2}')+$(free -m|awk '/^Swap:/{print $2}')))
+  if [ ${TOTAL_MEM_AVAILABLE} -le ${TOTAL_MEM_NEEDED} ]; then
+    echo "WARNING: Not enough free memory to start Jetty (about ${TOTAL_MEM_NEEDED}MB are needed). Trying anyway, but the application is likely to fail."
+  fi
+}
 
 
 ##################################################
@@ -351,6 +363,13 @@
 rm -f $JETTY_STATE
 
 ##################################################
+# Set default JVM parameters, to be
+# overriden by the configuration file
+##################################################
+JAVA_XMX=${JAVA_XMX:-1024}
+JAVA_MAXPERMSIZE=${JAVA_MAXPERMSIZE:-256}
+
+##################################################
 # Setup JAVA if unset
 ##################################################
 if [ -z "$JAVA" ]
@@ -421,6 +440,8 @@
   echo "JETTY_PID      =  $JETTY_PID"
   echo "JETTY_ARGS     =  $JETTY_ARGS"
   echo "CONFIGS        =  ${CONFIGS[*]}"
+  echo "JAVA_XMX       =  ${JAVA_XMX}"
+  echo "JAVA_MAXPERMSIZE =  ${JAVA_MAXPERMSIZE}"
   echo "JAVA_OPTIONS   =  ${JAVA_OPTIONS[*]}"
   echo "JAVA           =  $JAVA"
   echo "RUN_CMD        =  ${RUN_CMD}"
@@ -433,6 +454,10 @@
   start)
     UMASK="0027"
     echo "Setting umask to ${UMASK}"
+
+    # Checking if enough RAM is available for Jetty to use
+    checkAvailableRam $((${JAVA_XMX}+${JAVA_MAXPERMSIZE}))
+
     echo -n "Starting Jetty: "
 
     if (( NO_START )); then 
@@ -583,6 +608,8 @@
     echo "JETTY_LOGS     =  $JETTY_LOGS"
     echo "START_INI      =  $START_INI"
     echo "CONFIGS        =  ${CONFIGS[*]}"
+    echo "JAVA_XMX       =  ${JAVA_XMX}"
+    echo "JAVA_MAXPERMSIZE =  ${JAVA_MAXPERMSIZE}"
     echo "JAVA_OPTIONS   =  ${JAVA_OPTIONS[*]}"
     echo "JAVA           =  $JAVA"
     echo "CLASSPATH      =  $CLASSPATH"
