--- jetty/bin/jetty.sh	2017-10-20 21:18:40.570397097 +0200
+++ jetty/bin/jetty.sh_6	2017-10-20 21:31:56.110388623 +0200
@@ -313,6 +313,7 @@
 #####################################################
 # define start log location
 #####################################################
+JETTY_START_LOG="/var/log/rudder/webapp/java.log"
 if [ -z "$JETTY_START_LOG" ]
 then
   JETTY_START_LOG="$JETTY_RUN/$NAME-start.log"
@@ -507,11 +508,11 @@
         chown "$JETTY_USER" "$JETTY_PID"
         # FIXME: Broken solution: wordsplitting, pathname expansion, arbitrary command execution, etc.
         su - "$JETTY_USER" $SU_SHELL -c "
-          exec ${RUN_CMD[*]} start-log-file="$JETTY_START_LOG" > /dev/null &
+          exec ${RUN_CMD[*]} start-log-file="$JETTY_START_LOG" >${JETTY_START_LOG} 2>&1 &
           disown \$!
           echo \$! > '$JETTY_PID'"
       else
-        "${RUN_CMD[@]}" > /dev/null &
+        "${RUN_CMD[@]}" >${JETTY_START_LOG} 2>&1 &
         disown $!
         echo $! > "$JETTY_PID"
       fi
