--- jetty/bin/jetty.sh	2017-10-20 19:27:44.756967360 +0200
+++ jetty/bin/jetty.sh_4	2017-10-20 21:07:09.763737790 +0200
@@ -210,11 +210,10 @@
   ETC=$HOME/etc
 fi
 
-for CONFIG in {/etc,~/etc}/default/${NAME}{,9} $HOME/.${NAME}rc; do
-  if [ -f "$CONFIG" ] ; then
-    readConfig "$CONFIG"
-  fi
-done
+CONFIG=/etc/default/rudder-jetty
+if [ -f "$CONFIG" ] ; then
+  readConfig "$CONFIG"
+fi
 
 
 ##################################################
